File: Samples/BlinkID-UI-Sample-ObjC/BlinkID-UI-Sample-ObjC/ViewController.m
//
//  ViewController.m
//  BlinkID-UI-Sample-ObjC
//
//  Created by Jure Čular on 12/12/2018.
//  Copyright © 2018 Microblink. All rights reserved.
//

#import "ViewController.h"
#import <MicroBlink/MicroBlink.h>
#import <BlinkIDUI/BlinkIDUI.h>

@interface ViewController () <MBBlinkDelegate>

@property (nonatomic, nonnull) MBBlinkIDUI *blinkIdUI;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [[MBMicroblinkSDK sharedInstance] setLicenseResource:@"blinkid-license" withExtension:@"txt" inSubdirectory:nil forBundle:NSBundle.mainBundle];
    [self setupBlinkIDUI];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [self presentViewController:self.blinkIdUI.recognizerRunnerViewController animated:YES completion:nil];
}

- (void)setupBlinkIDUI {
    MBBlinkSettings.sharedInstance.frameGrabberMode = MBFrameGrabberModeNothing;
    MBBlinkSettings.sharedInstance.initialCountry = [[MBCountry alloc] initWithSupportedCountry:MBSupportedCountryUnitedStatesOfAmerica];
    MBBlinkSettings.sharedInstance.initialDocument = MBDocumentTypeDriverLicense;
    
    MBBlinkSettings.sharedInstance.shouldShowCancelButton = NO;
    
    // Change the colors
     UIColor.mb_primaryColor = UIColor.redColor;
     UIColor.mb_secondaryColor = UIColor.blackColor;
     UIColor.mb_tertiaryColor = UIColor.whiteColor;
     UIColor.mb_shadowColor = UIColor.whiteColor;
    
    self.blinkIdUI = [[MBBlinkIDUI alloc] init];
    self.blinkIdUI.delegate = self;
}

// MARK: - MBBlinkDelegate -

- (void)didScanEntireDocumentWithRecognitionResult:(MBRecognitionResult * _Nonnull)recognitionResult successFrame:(UIImage * _Nullable)successFrame {
    [self.blinkIdUI pauseScanning];
    
    UIAlertController *alerController = [UIAlertController alertControllerWithTitle:recognitionResult.resultTitle message:recognitionResult.resultEntries.description preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [self.blinkIdUI resumeScanning];
        [self.blinkIdUI restartScanning];
    }];
    
    [alerController addAction:okAction];
    
    [self.blinkIdUI.recognizerRunnerViewController presentViewController:alerController animated:YES completion:nil];
}

- (void)didScanFirstSideWithRecognitionResult:(MBRecognitionResult * _Nonnull)recognitionResult successFrame:(UIImage * _Nullable)successFrame {
}

- (void)didTapCancelButton {
    
}

- (void)didChangeDocumentWithNewDocument:(MBDocumentProvider *)newDocument forCountry:(MBCountry *)country {
    
}

- (void)didStartScanningWithState:(enum MBScanState)state {
    
}

@end

File: Samples/BlinkID-UI-Sample-Swift/BlinkID-UI-Sample-Swift/ViewController.swift
//
//  ViewController.swift
//  BlinkID-UI-Sample-Swift
//
//  Created by Jure Čular on 12/12/2018.
//  Copyright © 2018 Microblink. All rights reserved.
//

import UIKit
import MicroBlink
import BlinkIDUI

class ViewController: UIViewController {

    lazy var blinkIdUI = MBBlinkIDUI()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        MBMicroblinkSDK.sharedInstance().setLicenseResource("blinkid-license", withExtension: "txt", inSubdirectory: nil, for: Bundle.main)
        setupBlinkIDUI()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        present(blinkIdUI.recognizerRunnerViewController, animated: false, completion: nil)
    }
    
    func setupBlinkIDUI() {
        MBBlinkSettings.sharedInstance.frameGrabberMode = .nothing
        MBBlinkSettings.sharedInstance.initialCountry = MBCountry(supportedCountry: .unitedStatesOfAmerica)
        MBBlinkSettings.sharedInstance.initialDocument = .driverLicense
        
        MBBlinkSettings.sharedInstance.shouldShowCancelButton = false
        MBBlinkSettings.sharedInstance.shouldValidateDocuments = true
        // Change the colors
        // UIColor.mb_primaryColor = .red
        // UIColor.mb_secondaryColor = .black
        // UIColor.mb_tertiaryColor = .white
        // UIColor.mb_shadowColor = .white
        
        blinkIdUI.delegate = self
    }
}

extension ViewController: MBBlinkDelegate {
    func didScanEntireDocument(recognitionResult: MBRecognitionResult, successFrame: UIImage?) {
        blinkIdUI.pauseScanning()
        let alertController = UIAlertController(title: recognitionResult.resultTitle, message: recognitionResult.resultEntries.description, preferredStyle: .alert)
        let okAction = UIAlertAction(title: "OK", style: .default) { (_) in
            self.blinkIdUI.resumeScanning()
            self.blinkIdUI.restartScanning()
        }
        alertController.addAction(okAction)
        blinkIdUI.recognizerRunnerViewController.present(alertController, animated: true, completion: nil)
    }
    
    func didScanFirstSide(recognitionResult: MBRecognitionResult, successFrame: UIImage?) {
    }
}

